msrestazure/azure_active_directory.py

```diff
@@ -633,6 +639,21 @@ def get_token(self, resource):
         self.cache[resource] = token_entry
         return token_entry
 
+    def _sleep(self, time_to_wait, start_time):
+        """Sleep for time_to_wait or time remaining until timeout reached.
+
+        :param float time: Time to sleep in seconds
+        :param float start_time: Absolute time where polling started
+        :rtype: bool
+        :returns: True if timeout was used
+        """
+        if self.timeout:
```

Comment:
Does this handle when `timeout` is 0 so we only try once?

Reply:
Haha, that's interesting. Indeed if you say timeout 0, you actually say "try once only". You're right, right now it will assume there is no timeout.
Let me update that